<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Vignette for genecounteR</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Vignette for genecounteR</h1>



<div id="before-starting" class="section level2">
<h2>Before starting</h2>
<p>We recommend that users follow this vignette with the example data
before using on their own data sets. Feel free to copy and paste the
below code into a new R script when running on your own data. There are
several steps to follow which work best when followed sequentially.
Please install the libraries ‘devtools’ (for library installation) and
‘scales’ (for transparency on plots) before proceeding.</p>
</div>
<div id="install-the-library" class="section level2">
<h2>Install the library</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;MarkRChristie/genecounteR&quot;</span>)</span></code></pre></div>
</div>
<div id="load-the-library" class="section level2">
<h2>Load the library</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&#39;genecounteR&#39;</span>)</span></code></pre></div>
</div>
<div id="set-working-directory" class="section level2">
<h2>Set working directory</h2>
<p>Because you will be working with a medium sized dataset, we suggest
setting a working directory so you can later delete the file.</p>
<p>You can set the working directory directly or use the R package
‘here’. For the example we will just use the Downloads directory; in
practice you may want to create a specific directory:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;~/Downloads/&quot;</span>) <span class="co"># use something like this for Linux and Mac operating systems</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;C:/Users/username/Downloads&quot;</span>) <span class="co"># use something like this for Windows machines, where username should be substituted with your actual user name.</span></span></code></pre></div>
<p>or you can use the pacakge ‘here’ to create relative directories.
Learn more <a href="https://cloud.r-project.org/web/packages/here/index.html">here</a></p>
</div>
<div id="load-the-test-dataset" class="section level2">
<h2>Load the test dataset</h2>
<p>A test dataset called “TestData.txt” with 15 samples and 120,076
“genes” can be downloaded directly from the <a href="https://github.com/MarkRChristie/genecounteR/tree/main/inst/extdata">genecounter
github repo</a></p>
<p>Alternatively, you can download the file directly from Dropbox into
your current working directory:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">download.file</span>(<span class="st">&#39;https://www.dropbox.com/scl/fi/zcx4ou8wahov9wivw8hxq/TestData.txt?rlkey=jdzk0qlpixsw13y3thec1zhmr&amp;st=hcx6duqv&amp;dl=1&#39;</span>,</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>              <span class="at">destfile=</span><span class="st">&quot;TestData.txt&quot;</span>,</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>              <span class="at">method=</span><span class="st">&quot;auto&quot;</span>) <span class="co"># will fail if a file named TestData.txt already exists in your working directory</span></span></code></pre></div>
<p>Next, we load the file into R and perform some minimal formatting.
After formatting, the first column should be a gene name or gene id, the
second column the gene lengths, and each remaining column the gene
counts per sample (one column per sample). Each row should be a
gene/feature/isoform. The test dataset comes from a standard
featurecounts pipeline.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>gene.counts <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;TestData.txt&quot;</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">header=</span>T, <span class="at">row.names=</span><span class="dv">1</span>, <span class="at">check.names=</span>F)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="fu">colnames</span>(gene.counts)</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>dat <span class="ot">&lt;-</span> gene.counts[, <span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)]  <span class="co"># remove all columns except gene lengths and gene counts</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(dat), dat) <span class="co"># add unique gene ids; can use rownames if you want to preserve name</span></span></code></pre></div>
<p>When importing your own dataset as “dat” make sure that each row is a
gene and that<br />
the columns are formatted as (col1= gene id; col2 = gene length; col3 =
sample 1 counts; col4 = sample 2 counts etc.) Run head(gene.counts) and
head(dat) after running the above code to understand required data
format.</p>
</div>
<div id="zero-remover" class="section level2">
<h2>Zero Remover</h2>
<p>Remove all genes that have all zeros for counts; 2 plots created</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>percent.zeros.retained <span class="ot">&lt;-</span> <span class="fl">0.8</span> <span class="co"># percentage of samples allowed to have a count of zero; applied for each gene (row) separately</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">ZeroRemover</span>(dat, percent.zeros.retained)</span></code></pre></div>
</div>
<div id="sample-standardization" class="section level2">
<h2>Sample Standardization</h2>
<p>Standardize gene counts by the total number of reads per sample; 1
plot created</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">SampleStandardization</span>(dat)</span></code></pre></div>
</div>
<div id="gene-standardization" class="section level2">
<h2>Gene Standardization</h2>
<p>Standardize by residuals between gene length and total expression;
linear by default; creates 4 plots</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>dat  <span class="ot">&lt;-</span> <span class="fu">GeneStandardization</span>(dat)</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>ref2 <span class="ot">&lt;-</span> dat <span class="co"># for Matching Back after analyses are completed </span></span></code></pre></div>
<p>Users can import their own standardized data sets (using different
methods) at this step if so desired.</p>
</div>
<div id="visualize-expression" class="section level2">
<h2>Visualize Expression</h2>
<p>Retain top x percent of genes that equal or exceed a log mean gene
count, where x is set with “threshold”; illustrated on the single plot
that is generated.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>threshold <span class="ot">&lt;-</span> <span class="fl">0.15</span> <span class="co"># will retain top 15% of genes by log mean expression</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>dat       <span class="ot">&lt;-</span> <span class="fu">GeneCountDistributionPlot</span>(dat, threshold) </span></code></pre></div>
</div>
<div id="rank-order-expression" class="section level2">
<h2>Rank Order Expression</h2>
<p>Calculate rank-ordered expression within each gene; 3 plots
generated</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>deltas <span class="ot">&lt;-</span> <span class="fu">RankOrder</span>(dat) </span></code></pre></div>
</div>
<div id="boot-strap" class="section level2">
<h2>Boot Strap</h2>
<p>Calculate significance of each gene; 1 plot generated, significant
genes in red</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>alpha  <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>deltas <span class="ot">&lt;-</span> <span class="fu">BootStrap</span>(dat, deltas, alpha)</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>deltas[, <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>, <span class="fu">ncol</span>(deltas))] <span class="co"># visual results</span></span></code></pre></div>
</div>
<div id="match-back" class="section level2">
<h2>Match Back</h2>
<p>Match back to original data set, plot top genes, report gene IDs</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="fu">MatchBack</span>(deltas, dat, ref2)</span></code></pre></div>
</div>
<div id="run-all" class="section level2">
<h2>Run All</h2>
<p>If you would like to run all of the steps at once, you can wrap them
into a single function called RunAll and then simply call that function.
Generally not advised, but included below are all the R commands run
above without many comments, which may facilitate creating code that
works best for you!</p>
<p>Step 1: create function</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>RunAll <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>  <span class="fu">library</span>(<span class="st">&#39;genecounteR&#39;</span>)</span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a>  <span class="fu">setwd</span>(<span class="st">&quot;~/Downloads/&quot;</span>) </span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a>  gene.counts <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;TestData.txt&quot;</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">header=</span>T, <span class="at">row.names=</span><span class="dv">1</span>, <span class="at">check.names=</span>F)</span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a>  <span class="fu">colnames</span>(gene.counts)</span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a>  </span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a>  dat  <span class="ot">&lt;-</span> gene.counts[, <span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)]  <span class="co"># remove all columns except gene lengths and gene counts</span></span>
<span id="cb13-8"><a href="#cb13-8" tabindex="-1"></a>  dat  <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(dat), dat)</span>
<span id="cb13-9"><a href="#cb13-9" tabindex="-1"></a>  percent.zeros.retained <span class="ot">&lt;-</span> <span class="fl">0.8</span> <span class="co"># percentage of samples allowed to have a count of zero; applied for each gene (row) seperately</span></span>
<span id="cb13-10"><a href="#cb13-10" tabindex="-1"></a>  dat  <span class="ot">&lt;-</span> <span class="fu">ZeroRemover</span>(dat, percent.zeros.retained)</span>
<span id="cb13-11"><a href="#cb13-11" tabindex="-1"></a>  dat  <span class="ot">&lt;-</span> <span class="fu">SampleStandardization</span>(dat)</span>
<span id="cb13-12"><a href="#cb13-12" tabindex="-1"></a>  dat  <span class="ot">&lt;-</span> <span class="fu">GeneStandardization</span>(dat)</span>
<span id="cb13-13"><a href="#cb13-13" tabindex="-1"></a>  ref2 <span class="ot">&lt;-</span> dat <span class="co"># for Matching Back after analyses are completed </span></span>
<span id="cb13-14"><a href="#cb13-14" tabindex="-1"></a>  threshold <span class="ot">&lt;-</span> <span class="fl">0.15</span></span>
<span id="cb13-15"><a href="#cb13-15" tabindex="-1"></a>  dat    <span class="ot">&lt;-</span> <span class="fu">GeneCountDistributionPlot</span>(dat, threshold) </span>
<span id="cb13-16"><a href="#cb13-16" tabindex="-1"></a>  deltas <span class="ot">&lt;-</span> <span class="fu">RankOrder</span>(dat) </span>
<span id="cb13-17"><a href="#cb13-17" tabindex="-1"></a>  alpha  <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb13-18"><a href="#cb13-18" tabindex="-1"></a>  deltas <span class="ot">&lt;-</span> <span class="fu">BootStrap</span>(dat, deltas, alpha)</span>
<span id="cb13-19"><a href="#cb13-19" tabindex="-1"></a>  deltas[, <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">4</span>, <span class="fu">ncol</span>(deltas))] <span class="co"># see results table</span></span>
<span id="cb13-20"><a href="#cb13-20" tabindex="-1"></a>  <span class="fu">MatchBack</span>(deltas, dat, ref2)</span>
<span id="cb13-21"><a href="#cb13-21" tabindex="-1"></a>}</span></code></pre></div>
<p>Step 2: run function</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="fu">RunAll</span>()</span></code></pre></div>
</div>
<div id="notes" class="section level2">
<h2>Notes</h2>
<p>Program created by Mark Christie; contact at <a href="mailto:markchristie1500@gmail.com" class="email">markchristie1500@gmail.com</a><br />
Created in R 4.4.3<br />
genecounteR: a simple framework for identifying constituitively
expressed genes in large RNA-Seq data sets</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
